git-workflow.md

```diff
@@ -0,0 +1,77 @@
+# `git` Workflow
+
+## Terminology
+- `upstream` : This is the repo you want to contribute to. Ex: `github.com/konveyor/move2kube`
+- `origin` : This is your fork of the `upstream` repo. Ex: `github.com/myusername/move2kube`
+- `local` : This is a repo on your local machine (laptop/desktop). Usually this refers to the clone of your fork `origin`.
+
+## Setup
+This only needs to be done once:
+1. Fork the repo on Github.
+2. Clone your fork: `git clone <my fork url>`
+3. Add the `upstream` repo to the set of remote repos: `git remote add upstream <upstream url>`
+4. Check that you have both the `origin` and `upstream` set correctly: `git remote -v`
+
+## Debugging/Visibility
+When in doubt `git log --graph --all` and scroll around with the arrow keys.  
+This shows you the graph of commits and the commit each branch is pointing to.  
+If there are a lot of commits you can also add `--one-line` to make it easier to see.
+
+## Sync up
+Often our `local` repo and our fork `origin` will lag behind the main repo `upstream`.
+It is important to sync up with `upstream` before we rebase and submit a pull request on Github.
+
+1. Get the latest code from upstream: `git fetch upstream`
+2. Switch to the master branch: `git checkout master`
+3. Fast forward your local master branch to catch up with upstream: `git merge --ff-only upstream/master`
```

Comment:
How about, 'git rebase upstream/master'

Reply:
no you don't need to rebase.
A fast forward merge will simply move the `master` pointer a few commits forward to where `upstream/master` is at.